tests/platform/api/test_watchdog.py

```diff
@@ -0,0 +1,176 @@
+import os
+import time
+import logging
+import yaml
+import pytest
+from common.helpers.platform_api import watchdog
+
+logger = logging.getLogger(__name__)
+
+TEST_CONFIG_FILE = os.path.join(os.path.split(__file__)[0], "watchdog.yml")
+
+class TestWatchdogAPI(object):
+    ''' Hardware watchdog platform API test cases '''
+
+    @pytest.fixture(scope='function', autouse=True)
+    def watchdog_not_running(self, platform_api_conn):
+        ''' Fixture that automatically runs on each test case and
+        verifies that watchdog is not running before the test begins
+        and disables it after the test ends'''
+
+        assert not watchdog.is_armed(platform_api_conn)
+
+        try:
+            yield
+        finally:
+            watchdog.disarm(platform_api_conn)
+
+    @pytest.fixture(scope='module')
+    def conf(self, request, duthost):
+        ''' Reads the watchdog test configuration file @TEST_CONFIG_FILE and
+        results in a dictionary which holds parameters for test '''
+
+        test_config = None
+        with open(TEST_CONFIG_FILE) as stream:
+            test_config = yaml.safe_load(stream)['default']
+
+        platform = duthost.facts['platform']
+        hwsku = duthost.facts['hwsku']
+
+        if platform in test_config and 'default' in test_config[platform]:
+            test_config.update(test_config[platform]['default'])
+
+        if platform in test_config and hwsku in test_config[platform]:
+            test_config.update(test_config[platform][hwsku])
+
+        return test_config
+
+
+    def test_arm_disarm_states(self, testbed_devices, platform_api_conn, conf):
+        ''' arm watchdog with a valid timeout value, verify it is in armed state,
+        disarm watchdog and verify it is in disarmed state
+        '''
+
+        duthost = testbed_devices['dut']
+        localhost = testbed_devices['localhost']
+
+        test_timeout = conf['valid_timeout']
+        actual_timeout = watchdog.arm(platform_api_conn, test_timeout)
+
+        assert actual_timeout != -1
+        assert actual_timeout >= test_timeout
+        assert watchdog.is_armed(platform_api_conn)
+
+        assert watchdog.disarm(platform_api_conn)
+        assert not watchdog.is_armed(platform_api_conn)
```

Comment:
it is assumed disarm is done on the same time it is requested. Is it valid assumption for different vendors or there is some brief timeout need to sleep before you can actually verify watchdog is not armed?

Reply:
The remote API and platform API is blocking, not asynchronous, if platform API implementation needs some brief time it should be coded in platform API implementation